exercises/components-in-depth/README.md

```diff
@@ -1,93 +1,71 @@
-- **Reset timer:** Resets the timer
-
-Using `<ng-content>` with `select`, allow each button to receive its own custom label
+Use the `angular-cli` to `generate` a `button-grouping` and a `counter-display` 
+component.
 
 ## Step 3
 
-Add an `@Input` interval property, and a time property both of type `number`
+In `button-grouping.component.ts`, add the following `@Input` property
+- counterValue
 
-- **interval** Acts as the step interval for the timer
-- **time:** Will store the current callculated time
-- **timerId:** Will store the current timer id
+and the following `@Output` EventEmitters
+- onIncrement
+- onDecrement
 
 ## Step 4
 
-Add 4 `@Output` `EventEmitter`s
+Implement the `increment` method to emit the `onIncrement` event containing the 
+`counterValue` incremented by 1.
 
-- **onStart:** Emits when the timer starts
-- **onStop:** Emits when the timer stops
-- **onInterval:** Emits on every timer interval
-- **onReset:** Emits when the timer is reset
+Implement the `decrement` method to emit the `onDecrement` event containing the 
+`counterValue` decremented by 1.
 
 ## Step 5
 
-Implement a `start` method to start a timer using `window.setInterval` and store the returned id in the `timerId` property.
+Create two buttons in `button-grouping.component.html` that will call our 
+`increment` and `decrement` methods when pressed.
 
-The timer should use the `interval` input value as its interval and should update the `time` property with the latest time.
-The timer should also emit the latest time at every interval using the `onInterval` output.
-
-Current time can be computed as `time += interval / 1000`
-
-Optional: The method should also emit `time` if a timer hasn't yet been started using the `onStart` output.
+Use `ng-content` inside the buttons such that we can use projection to pass in
+custom button text.
 
 ## Step 6
 
-Using `window.clearInterval` implement a `stop` method that will stop the current running timer using the stored `timerId` id.
-
-Optional: Emit the current time using the `onStop` output.
+In `counter-display.component.ts`:
+- create a `defaultCounterValue` variable with a value of 10
+- create a `counterValue` variable which will keep track of the current value
+- implement a `setDefaultValue` method which will update `counterValue` to the 
+default value
+- call the `setDefaultValue` in the `OnInit` hook to instantiate the value of 
+`counterValue`
 
 ## Step 7
 
-Implement a `reset` that will reset the `time` to `0` and emit the current time using the `onReset` output
-
-## Step 8
-
-Create a `<app-fare-calculator>` component which will use `<app-timer>` in order to calculate fare based on a given rate.
-
-## Step 9
-
-Add 2 `@Input`s `rate`, and `interval`, and one property `time`
+Place the following snippet into `counter-display.component.html`.
+```html
+<input type="number" [(ngModel)]="defaultCounterValue">
+<button (click)="setDefaultValue()">
+  Set default value
+</button>
```

Comment:
@jhnching What do you think if we remove the angular specific part and ask the user to add it?

So instead of showing:

`<input type="number" [(ngModel)]="defaultCounterValue">`

We show this:

`<input type="number" >`

in the exercise and ask them to complete it with angular specific code.